doc/kb_sdk_make_ui.md

```diff
@@ -0,0 +1,169 @@
+# <A NAME="top"></A>![alt text](https://avatars2.githubusercontent.com/u/1263946?v=3&s=84 "KBase") [KBase SDK](../README.md)
+
+1. [Install SDK Dependencies](kb_sdk_dependencies.md)
+2. [Install SDK with Docker](kb_sdk_dockerized_install.md)
+3. [Create Module](kb_sdk_create_module.md)
+4. [Specify Module and Method(s)](kb_sdk_edit_module.md)
+5. [Implement Method(s)](kb_sdk_impl_methods.md)
+6. **Specify User Interface**
+7. [Locally Test Module and Method(s)](kb_sdk_local_test_module.md)
+8. [Register Module](kb_sdk_register_module.md)
+9. [Test in KBase](kb_sdk_test_in_kbase.md)
+10. [Complete Module Info](kb_sdk_complete_module_info.md)
+11. [Deploy](kb_sdk_deploy.md)
+
+### <A NAME="ui-widget"></A>6. Specify User Interface
+
+Control of Narrative interaction is accomplished in files in the ui/narrative/methods/<MyMethod> directory.
+
+#### A. Configure the input interface.
+
+The input options are specified in the "parameters" section of the spec.json file. In the following example, the user
+supplies two required parameters, an input name and an output name. By specifying a 'valid_ws_types' the user will
+be present a searchable dropdown of objects that match the specified type. By passing "is_output_name", the user is
+warned if a name will overwrite an existing object or if the name contains invalid characters.
+
+```
+	"parameters": [ 
+		{
+			"id": "read_library_name",
+			"optional": false,
+			"advanced": false,
+			"allow_multiple": false,
+			"default_values": [ "" ],
+			"field_type": "text",
+			"text_options": {
+				"valid_ws_types": ["KBaseAssembly.PairedEndLibrary","KBaseFile.PairedEndLibrary"]
+			}
+		},
+		{
+		    "id" : "output_contigset_name",
+		    "optional" : false,
+		    "advanced" : false,
+		    "allow_multiple" : false,
+		    "default_values" : [ "MegaHit.contigs" ],
+		    "field_type" : "text",
+		    "text_options" : {
+		     	"valid_ws_types" : [ "KBaseGenomes.ContigSet" ],
+		    	"is_output_name":true
+		    }
+		}
+	],
+```
+Another common input is a dropdown which is demonstrated below. For each option, the "value" is what will be passed to
+the script while the UI Name and Display is what the user will see. In this example, the parameter is hidden be default 
+because "advanced" is true.
+```
+{
+            "id": "prune",
+            "optional": false,
+            "advanced": true,
+            "allow_multiple": false,
+            "default_values": [ "biochemistry" ],
+            "field_type": "dropdown",
+            "dropdown_options": {
+                "options": [
+                    {
+                        "value": "biochemistry",
+                        "display": "Known Biochemistry",
+                        "id": "biochemistry",
+                        "ui_name": "Known Biochemistry"
```

Comment:
I'm pretty sure that only `ui_name` is used at this point, and `display` is redundant. I'll have to test a few things to check, though.